dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/master/runner/MasterBaseTaskExecThread.java

```diff
@@ -247,30 +252,27 @@ public Boolean call() throws Exception {
      * @return log path
      */
     public String getTaskLogPath(TaskInstance task) {
-        String logPath;
-        try{
-            String baseLog = ((TaskLogDiscriminator) ((SiftingAppender) ((LoggerContext) LoggerFactory.getILoggerFactory())
-                    .getLogger("ROOT")
-                    .getAppender("TASKLOGFILE"))
-                    .getDiscriminator()).getLogBase();
-            if (baseLog.startsWith(Constants.SINGLE_SLASH)){
-                logPath =  baseLog + Constants.SINGLE_SLASH +
-                        task.getProcessDefinitionId() + Constants.SINGLE_SLASH  +
-                        task.getProcessInstanceId() + Constants.SINGLE_SLASH  +
-                        task.getId() + ".log";
-            }else{
-                logPath = System.getProperty("user.dir") + Constants.SINGLE_SLASH +
-                        baseLog +  Constants.SINGLE_SLASH +
-                        task.getProcessDefinitionId() + Constants.SINGLE_SLASH  +
-                        task.getProcessInstanceId() + Constants.SINGLE_SLASH  +
+        try {
+            // Optional.map will be skipped if null
+            String baseLog = Optional.of(LoggerFactory.getILoggerFactory().getLogger("ROOT"))
+                    .map(e -> (AppenderAttachable<ILoggingEvent>) e)
+                    .map(e -> (SiftingAppender) (e.getAppender("TASKLOGFILE")))
+                    .map(e -> ((TaskLogDiscriminator) (e.getDiscriminator())))
+                    .map(e -> e.getLogBase())
+                    .orElse(null);
+            if (baseLog != null) {
+                if (!baseLog.startsWith(Constants.SINGLE_SLASH)) {
+                    // complement the relative path
+                    baseLog = System.getProperty("user.dir") + Constants.SINGLE_SLASH + baseLog;
+                }
+                return baseLog + Constants.SINGLE_SLASH +
+                        task.getProcessDefinitionId() + Constants.SINGLE_SLASH +
+                        task.getProcessInstanceId() + Constants.SINGLE_SLASH +
```

Comment:
```suggestion
        @SuppressWarnings("unchecked")
    public String getTaskLogPath(TaskInstance task) {
        try {
            // Optional.map will be skipped if null
            String baseLog = Optional.of(LoggerFactory.getILoggerFactory().getLogger("ROOT"))
                    .map(e -> (AppenderAttachable<ILoggingEvent>) e)
                    .map(e -> (SiftingAppender) (e.getAppender("TASKLOGFILE")))
                    .map(e -> ((TaskLogDiscriminator) (e.getDiscriminator())))
                    .map(TaskLogDiscriminator::getLogBase)
                    .orElse(null);
            if (baseLog != null) {
                if (!baseLog.startsWith(Constants.SINGLE_SLASH)) {
                    // complement the relative path
                    baseLog = System.getProperty("user.dir") + Constants.SINGLE_SLASH + baseLog;
                }
                return baseLog + Constants.SINGLE_SLASH
                        + task.getProcessDefinitionId() + Constants.SINGLE_SLASH
                        + task.getProcessInstanceId() + Constants.SINGLE_SLASH
                        + task.getId() + ".log";
            }
        } catch (Exception e) {
            logger.error("logger", e);
        }
        return "";
    }
```

Reply:
Good solution about this NPE situation.
This is the checkstyle format result, please update.
And the sonar fail is because the lower 33.3% test coverage of new code, please resolve.